src/Apps/Order/Routes/__tests__/Payment.test.tsx

```diff
@@ -0,0 +1,217 @@
+import { mount, shallow } from "enzyme"
+import React from "react"
+
+import { OrderWithShippingDetails } from "Apps/__test__/Fixtures/Order"
+import { Checkbox } from "../../../../Styleguide/Elements"
+import { AddressForm } from "../../Components/AddressForm"
+import { ContinueButton, PaymentProps, PaymentRoute } from "../Payment"
+
+jest.mock("react-relay", () => ({
+  commitMutation: jest.fn(),
+  createFragmentContainer: component => component,
+}))
+
+import { Sans } from "@artsy/palette"
+import { commitMutation, RelayProp } from "react-relay"
+import { creatingCreditCardSuccess } from "../../../__test__/Fixtures/MutationResults/createCreditCard"
+import { settingOrderPaymentSuccess } from "../../../__test__/Fixtures/MutationResults/setOrderPayment"
+
+const mutationMock = commitMutation as jest.Mock<any>
+
+describe("Payment", () => {
+  let stripeMock
+
+  const mountPaymentRoute = props => {
+    const paymentRouteComponent = shallow(<PaymentRoute {...props} />)
+    const responsiveComponent = shallow(
+      paymentRouteComponent
+        .childAt(2)
+        .props()
+        .children({ xs: false })
+    ) as any
+    const twoColumnLayout = shallow(
+      responsiveComponent.props().children({ xs: false })
+    )
+
+    return {
+      paymentRouteComponent,
+      responsiveComponent,
+      twoColumnLayout,
+    }
```

Comment:
This is the questionable function that shallow-mounts the component three times. We don't want to mount the component entirely otherwise Stripe components will fail due to the lack of `context` values that are supposed to be injected. The library's test itself does seem to [test them with context values injected properly](https://github.com/stripe/react-stripe-elements/blob/732dd3ae7c64f171218064af012fbc2b13842482/src/components/Element.test.js#L38), but I couldn't figure out why this was not working in our setup.

Reply:
Have you tried an approach similar to https://github.com/artsy/reaction/blob/master/src/Utils/RelayStubProvider.ts?